azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/__init__.py

```diff
@@ -33,51 +45,110 @@ def configure_azure_monitor(**kwargs):
 
-    resource = None
-    if not disable_logging or not disable_tracing:
-        resource = Resource.create(
-            {
-                ResourceAttributes.SERVICE_NAME: service_name,
-                ResourceAttributes.SERVICE_NAMESPACE: service_namespace,
-                ResourceAttributes.SERVICE_INSTANCE_ID: service_instance_id,
-            }
-        )
-    if not disable_logging:
-        logger_provider = LoggerProvider(resource=resource)
-        set_logger_provider(logger_provider)
-        log_exporter = AzureMonitorLogExporter(**kwargs)
-        log_record_processor = BatchLogRecordProcessor(
-            log_exporter,
-            export_timeout_millis=logging_export_interval_millis,
-        )
-        get_logger_provider().add_log_record_processor(log_record_processor)
-        handler = LoggingHandler(
-            level=logging_level, logger_provider=get_logger_provider()
-        )
-        getLogger().addHandler(handler)
-    if not disable_tracing:
-        tracer_provider = TracerProvider(
-            sampler=ApplicationInsightsSampler(sampling_ratio=sampling_ratio),
-            resource=resource,
-        )
-        set_tracer_provider(tracer_provider)
-        trace_exporter = AzureMonitorTraceExporter(**kwargs)
-        span_processor = BatchSpanProcessor(
-            trace_exporter,
-            export_timeout_millis=tracing_export_interval_millis,
-        )
-        get_tracer_provider().add_span_processor(span_processor)
+
+def _setup_logging(resource: Resource, configurations: Dict[str, Any]):
+    logging_level = configurations.get("logging_level", NOTSET)
+    logging_export_interval_millis = configurations.get(
+        "logging_export_interval_millis", 30000
+    )
+    logger_provider = LoggerProvider(resource=resource)
+    set_logger_provider(logger_provider)
+    log_exporter = AzureMonitorLogExporter(**configurations)
+    log_record_processor = BatchLogRecordProcessor(
+        log_exporter,
+        export_timeout_millis=logging_export_interval_millis,
+    )
+    get_logger_provider().add_log_record_processor(log_record_processor)
+    handler = LoggingHandler(
+        level=logging_level, logger_provider=get_logger_provider()
+    )
+    getLogger().addHandler(handler)
+
+
+def _setup_instrumentations(configurations: Dict[str, Any]):
+    instrumentations = configurations.get("instrumentations", [])
+    for lib_name in instrumentations:
+        if lib_name in _SUPPORTED_INSTRUMENTED_LIBRARIES:
+            try:
+                importlib.import_module(lib_name)
+            except ImportError:
+                _logger.warning(
+                    "Unable to import %s. Please make sure it is installed.",
+                    lib_name,
+                )
+                continue
+            instr_lib_name = "opentelemetry.instrumentation." + lib_name
+            try:
+                module = importlib.import_module(instr_lib_name)
```

Comment:
This seems like it will work fine. But do you have thoughts on using entry points and get_dist_dependency_conflicts like sitecustomize does for consistency?

Reply:
That's probably a good idea. Let's merge this functionality for now and see if we want to change to using entry points.

Reply:
https://github.com/microsoft/ApplicationInsights-Python/issues/229